Tests/NIOHTTP2Tests/TestUtilities.swift

```diff
@@ -63,6 +63,60 @@ extension XCTestCase {
         } while operated
     }
 
+    /// Have two `EmbeddedChannel` objects send and receive data from each and throw errors other until
+    /// they make no forward progress.
+    func interactInMemoryExpectingErrors(
+        _ first: EmbeddedChannel,
+        _ second: EmbeddedChannel,
+        file: StaticString = #filePath,
+        line: UInt = #line,
+        _ errorHandler: (_ error: any Error) -> Void = { _ in }
+    ) {
+        var operated: Bool
+        var encounteredError = false
+
+        func readBytesFromChannel(_ channel: EmbeddedChannel) -> ByteBuffer? {
+            guard let data = try? assertNoThrowWithValue(channel.readOutbound(as: IOData.self)) else {
+                return nil
+            }
+            switch data {
+            case .byteBuffer(let b):
+                return b
+            case .fileRegion(let f):
+                return f.asByteBuffer(allocator: channel.allocator)
+            }
+        }
+
+        repeat {
+            operated = false
+
+            if let data = readBytesFromChannel(first) {
+                operated = true
+
+                do {
+                    try second.writeInbound(data)
+                } catch(let error) {
```

Comment:
No need for the `let error`, it's implicitly captured.

```suggestion
                } catch {
```